src/main/java/com/linecorp/armeria/common/ServiceInvocationContext.java

```diff
@@ -149,6 +150,28 @@ public final EventLoop eventLoop() {
     }
 
     /**
+     * Returns a {@link Executor} that will make sure this invocation is set
+     * as the current invocation before executing a callback. This should
+     * almost always be used for executing asynchronous callbacks in service
+     * code to make sure features that require the invocation context work
+     * properly. Most asynchronous libraries like
+     * {@link java.util.concurrent.CompletableFuture} provide methods that
+     * accept an {@link Executor} to run callbacks on.
+     */
+    public final Executor asyncCallbackExecutor() {
+        return command -> {
+            eventLoop().execute(() -> {
+                setCurrent(this);
```

Comment:
Let me know if I should check the current context or something. As far as I would expect, eventLoop.execute() is guaranteed to execute the task at some time in the future after the current calling code has returned since it's all just one thread, meaning there's no way this could be set at this time.